engine/src/flutter/lib/web_ui/lib/src/engine/semantics/label_and_value.dart

```diff
@@ -446,64 +447,129 @@ class LabelAndValue extends SemanticBehavior {
+        owner.removeAttribute('aria-valuetext');
+      } else {
+        owner.setAttribute('aria-valuetext', value);
+        owner.removeAttribute('aria-valuenow');
+      }
+    } else {
+      owner.removeAttribute('aria-valuenow');
+      owner.removeAttribute('aria-valuetext');
+    }
   }
 
-  LabelRepresentationBehavior? _representation;
-
-  /// Return the representation that should be used based on the current
-  /// parameters of the semantic node.
-  ///
-  /// If the node has children always use an `aria-label`. Using extra child
-  /// nodes to represent the label will cause layout shifts and confuse the
-  /// screen reader. If the are no children, use the representation preferred
-  /// by the role.
-  LabelRepresentationBehavior _getEffectiveRepresentation() {
-    final LabelRepresentation effectiveRepresentation =
-        semanticsObject.hasChildren ? LabelRepresentation.ariaLabel : preferredRepresentation;
-
-    LabelRepresentationBehavior? representation = _representation;
-    if (representation == null || representation.kind != effectiveRepresentation) {
-      representation?.cleanUp();
-      _representation = representation = effectiveRepresentation.createBehavior(owner);
+  String? _combineHintAndTooltip() {
+    final hint = semanticsObject.hint;
+    final tooltip = semanticsObject.hasTooltip ? semanticsObject.tooltip : null;
+    if (hint != null && tooltip != null && hint.trim().isNotEmpty && tooltip.trim().isNotEmpty) {
+      return '$hint $tooltip';
+    }
+    if (hint != null && hint.trim().isNotEmpty) {
+      return hint;
     }
-    return representation;
+    if (tooltip != null && tooltip.trim().isNotEmpty) {
+      return tooltip;
+    }
+    return null;
   }
 
-  /// Computes the final label to be assigned to the node.
-  ///
-  /// The label is a concatenation of tooltip, label, hint, and value, whichever
-  /// combination is present.
-  String? _computeLabel() {
-    // If the node is incrementable the value is reported to the browser via
-    // the respective role. We do not need to also render it again here.
-    final bool shouldDisplayValue = !semanticsObject.isIncrementable && semanticsObject.hasValue;
-
-    return computeDomSemanticsLabel(
-      tooltip: semanticsObject.hasTooltip ? semanticsObject.tooltip : null,
-      label: semanticsObject.hasLabel ? semanticsObject.label : null,
-      hint: semanticsObject.hint,
-      value: shouldDisplayValue ? semanticsObject.value : null,
-    );
+  void _ensureDescribedBy(String description) {
+    _describedById ??= 'flt-hint-${semanticsObject.id}';
+    _describedBySpan ??= domDocument.createElement('span');
+    _describedBySpan!.id = _describedById!;
+    _describedBySpan!.setAttribute('hidden', '');
+    _describedBySpan!.text = description;
+    // Attach as a child of the semantics element, not document.body
+    if (_describedBySpan!.parentNode != owner.element) {
+      owner.element.append(_describedBySpan!);
```

Comment:
The condition makes it seem possible that the span was appended to another element, and I don't see anywhere in the code that the span is getting attached to a different element.

What you are trying to do is check whether the span has already been inserted or not. If that's the case, let's make the condition more explicit:
```suggestion
    if (!_describedBySpan!.isConnected) {
      owner.element.append(_describedBySpan!);
```